pkgs/applications/science/chemistry/xcrysden/default.nix

```diff
@@ -0,0 +1,66 @@
+{ lib
+, fetchurl
+, fftw
+, gcc8Stdenv
+, gfortran
+, libGL
+, libGLU
+, tcl
+, tk
+, togl
+, util-linux
+, xorg
+}:
+
+let
+  bwidget = rec {
+    ver = "1.9.13";
+    tgz = "bwidget-${ver}.tar.gz";
+    src = fetchurl {
+      url = "http://sourceforge.net/projects/tcllib/files/BWidget/${ver}/${tgz}";
+      sha256 = "sha256-dtj0IoDnFgJCGG0SQ3lJgw6r1QCabBT059ug9mFAOoE=";
+    };
+  };
+
+in
+
+
+gcc8Stdenv.mkDerivation rec {
+  pname = "xcrysden";
+  version = "1.6.2";
+
+  src = fetchurl {
+    url = "http://www.xcrysden.org/download/xcrysden-${version}.tar.gz";
+    sha256 = "sha256-gRc27lmL7BpbQn/RDk4GOjDdfK2ulqQ6ULNs6QpPUD8=";
+  };
+
+  buildInputs = [
+    fftw
+    gfortran
+    libGL
+    libGLU
+    tcl
+    tk.dev
+    togl
+    xorg.libXmu
+  ];
+
+  configurePhase = ''
+    cp ${bwidget.src} external/src/${bwidget.tgz}
```

Comment:
Maybe you could add a comment here that the `bwidget` source code is actually required by the xcrysden build system?  (That is to say, it is not used as just a dependency, but the actual source code is required)

Reply:
Done: https://github.com/NixOS/nixpkgs/pull/218239/commits/66a91f278a97e4fb991d1411a6066fca40a5d354#diff-f59da3f9121929af98b7e2d8fd1b933b7e2a8fdbd2c6f2b5bcb338685f4d292eR15 (rebased)

Reply:
How about adding `bwidget` in `development/libraries/` as a derivation then ?

Reply:
Ho please, ignore this, I just noticed that `bwidget` is a set, not a derivation.